de.dlr.sc.virsat.model.extension.requirements.ui/src/de/dlr/sc/virsat/model/extension/requirements/ui/wizard/CsvTypeReviewPage.java

```diff
@@ -0,0 +1,330 @@
+	}
+
+	/**
+	 * Get the mapping of columns to their attribute in specified requirement type
+	 * 
+	 * @return a map of the column index to their attribute definition type
+	 */
+	public Map<Integer, RequirementAttribute> getAttributeMapping() {
+
+		for (TableItem item : tableItems) {
+			Integer index = tableItems.indexOf(item);
+
+			if (item.getChecked()) {
+				if (isNewType) {
+					String selectedType = editors.get(index).getText();
+
+					RequirementAttribute correspondingAtt = requirementType.getAttributes().get(index);
+
+					// Check if type is new and valid and then add it to the new requirement type
+					if (ATTRIBUTE_TYPES.contains(selectedType) && !selectedType.equals(correspondingAtt.getType())) {
+						correspondingAtt.setType(selectedType);
+					}
+
+					// Add corresponding attribute to mapping
+					mapColumnIndexToAttributeType.put(index, correspondingAtt);
+
+				} else {
+					String inputString = editors.get(index).getText();
+					
+					RequirementAttribute selectedAttribute = getSelectedAttribute(inputString);
+					if (selectedAttribute != null) {
+						mapColumnIndexToAttributeType.put(index, selectedAttribute);
+					}
+
+				}
+			} 
+
+		}
+
+		return mapColumnIndexToAttributeType;
+	}
+
+	/**
+	 * Get the configured requirement type for the import
+	 * 
+	 * @return the requirement type
+	 */
+	public RequirementType getRequirementType() {
+		if (isNewType) {
+			for (RequirementAttribute att : requirementType.getAttributes()) {
+				if (!mapColumnIndexToAttributeType.containsValue(att)) {
+					requirementType.getAttributes().remove(att);
+				}
+			}
+		}
+		return requirementType;
+	}
+
+	/**
+	 * Parse the attribute from the editor input string
+	 * 
+	 * @param inputString
+	 *            the editor input string
+	 * @return the attibute
+	 */
+	private RequirementAttribute getSelectedAttribute(String inputString) {
+
+		// Remove context indication from attribute
+		inputString = inputString.replace(requirementType.getName() + TYPE_LABEL_REQUIREMENT_ATTRIBUTE_SEPARATOR, "");
+
+		for (RequirementAttribute att : requirementType.getAttributes()) {
+
+			if (att.getName().equals(inputString)) {
+				return att;
+			}
+
+		}
+
+		return null;
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see org.eclipse.swt.events.SelectionListener#widgetSelected(org.eclipse.swt.
+	 * events.SelectionEvent)
+	 */
+	@Override
+	public void widgetSelected(SelectionEvent e) {
+		if (e.getSource().equals(table) && e.detail == SWT.CHECK) {
+			TableItem item = (TableItem) e.item;
+			CCombo editor = editors.get(tableItems.indexOf(item));
+			if (item.getChecked()) {
+				editor.setEditable(true);
+			} else {
+				editor.setEditable(false);
+			}
+		}
+		
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see
+	 * org.eclipse.swt.events.SelectionListener#widgetDefaultSelected(org.eclipse.
+	 * swt.events.SelectionEvent)
+	 */
+	@Override
+	public void widgetDefaultSelected(SelectionEvent e) {
+		
+	}
+
+	/* (non-Javadoc)

```

Comment:
Whats up with the weird javadoc (non-javadcoc?) comments here?

Reply:
It's generated methods from eclipse... Method overwrites from super classes, getter/ setters and delegate forwardings and so on... Eclipse can generate quite some stuff... But its adding these comments.